erroranalysis/erroranalysis/_internal/cohort_filter.py

```diff
@@ -26,259 +29,346 @@ def filter_from_cohort(analyzer, filters, composite_filters):
+        self.dataset = dataset
+        self.features = features
+        self.categorical_features = categorical_features
+        self.categories = categories
+        self.true_y = true_y
+        self.pred_y = pred_y
+        self.model_task = model_task
+        self.classes = classes
 
-def add_filter_cols(analyzer, df, filters, true_y):
-    """Adds special columns to the dataset for filtering and postprocessing.
+    def filter_data_from_cohort(self, filters, composite_filters,
+                                original_data_columns=False):
+        """Filters the dataset on the model based on the specified filters.
 
-    :param analyzer: The error analyzer.
-    :type: BaseAnalyzer
-    :param filters: The filters.
-    :type filters: list[dict]
-    :param true_y: The true labels.
-    :type true_y: list
-    """
-    has_classification_outcome = filters_has_classification_outcome(analyzer,
-                                                                    filters)
-    if isinstance(true_y, str):
-        df.rename(columns={true_y: TRUE_Y})
-    else:
-        df[TRUE_Y] = true_y
-    is_model_analyzer = hasattr(analyzer, MODEL)
-    if not is_model_analyzer:
-        df[PRED_Y] = analyzer.pred_y
-    if not is_spark(df):
-        df[ROW_INDEX] = np.arange(0, len(true_y))
-    if has_classification_outcome:
-        if PRED_Y in df:
-            pred_y = df[PRED_Y]
-        else:
-            # calculate directly via prediction on model
-            pred_y = analyzer.model.predict(df.drop(columns=[TRUE_Y,
-                                                             ROW_INDEX]))
-        classes = get_ordered_classes(analyzer.classes, true_y, pred_y)
-        # calculate classification outcome and add to df
-        df[CLASSIFICATION_OUTCOME] = compute_classification_outcome_data(
-            true_y, pred_y, classes)
-
-
-def compute_classification_outcome_data(true_y, pred_y, classes):
-    """Creates the classification outcome data.
-
-    :param true_y: The true labels.
-    :type true_y: list or numpy.ndarray
-    :param pred_y: The predicted labels.
-    :type pred_y: list or numpy.ndarray
-    :param classes: The classes.
-    :type classes: list
-    :return: The classification outcome data.
-    :rtype: list
-    """
-    classification_outcome = []
-    if not isinstance(pred_y, np.ndarray):
-        pred_y = np.array(pred_y)
-    if not isinstance(true_y, np.ndarray):
-        true_y = np.array(true_y)
-    for i in range(len(true_y)):
-        if true_y[i] == pred_y[i]:
-            if true_y[i] == classes[0]:
-                # True positive == 0
-                classification_outcome.append(0)
+        :param filters: The filters.
+        :type filters: list[dict]
+        :param composite_filters: The composite filters.
+        :type composite_filters: list[dict]
+        :return: The filtered dataset converted to a pandas DataFrame.
+        :param original_data_columns: Whether to just include the original
```

Comment:
nit: maybe include_original_columns might be a better variable name for a boolean type?  original_data_columns makes me think that I need the list of columns themselves.

Reply:
same above on line 22

Reply:
Thanks for the suggestion. Changed it to include_original_columns_only.